src/xcode/ENA/ENA/Source/Workers/Keychain/KeychainHelper.swift

```diff
@@ -76,43 +76,10 @@ struct KeychainHelper {
 		return nil
 	}
 
-
-	/// Generates and stores a new random database key
-	/// - Throws: a `KeychainError` in case the generation or database save fails
-	/// - Returns: the newly created key
-	func generateDatabaseKey() throws -> String {
-		var bytes = [UInt8](repeating: 0, count: 32)
-		let result = SecRandomCopyBytes(kSecRandomDefault, bytes.count, &bytes)
-		guard result == errSecSuccess else {
-			Log.error("Error creating random bytes.", log: .api)
-			throw KeychainError.keyGenerationFail
-		}
-
-		let key = "x'\(Data(bytes).hexEncodedString())'"
-		try saveToKeychain(key: SecureStore.keychainDatabaseKey, data: Data(key.utf8))
-		return key
-	}
-
-	/// Generates and stores a new random database key
-	/// - Throws: a `KeychainError` in case the generation or database save fails
-	/// - Returns: the newly created key
-	func generateContactDiaryDatabaseKey() throws -> String {
-		var bytes = [UInt8](repeating: 0, count: 32)
-		let result = SecRandomCopyBytes(kSecRandomDefault, bytes.count, &bytes)
-		guard result == errSecSuccess else {
-			Log.error("Error creating random bytes.", log: .api)
-			throw KeychainError.keyGenerationFail
-		}
-
-		let key = "x'\(Data(bytes).hexEncodedString())'"
-		try saveToKeychain(key: ContactDiaryStore.encryptionKeyKey, data: Data(key.utf8))
-		return key
-	}
-
 	/// Generates and stores a new random database key
 	/// - Throws: a `KeychainError` in case the generation or database save fails
 	/// - Returns: the newly created key
-	func generateEventDatabaseKey() throws -> String {
+	func generateDatabaseKey(persistAtKeychainKey keychainKey: String) throws -> String {
```

Comment:
```suggestion
	func generateDatabaseKey(persistForKeychainKey keychainKey: String) throws -> String {
```
`At` sounded to me more like I have to pass an url/location and not a key.

Reply:
renamed